subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/InputPropertiesSerializer.java

```diff
@@ -17,25 +17,30 @@
 package org.gradle.api.internal.changedetection.state;
 
 import org.gradle.api.GradleException;
-import org.gradle.internal.serialize.*;
+import org.gradle.internal.serialize.BaseSerializerFactory;
+import org.gradle.internal.serialize.Decoder;
+import org.gradle.internal.serialize.DefaultSerializer;
+import org.gradle.internal.serialize.Encoder;
+import org.gradle.internal.serialize.MapSerializer;
+import org.gradle.internal.serialize.Serializer;
 
 import java.util.Map;
 
 import static java.lang.String.format;
 
-class InputPropertiesSerializer implements Serializer<Map<String, Object>> {
+class InputPropertiesSerializer implements Serializer<Map<String, InputProperty>> {
 
-    private final MapSerializer<String, Object> serializer;
+    private final MapSerializer<String, InputProperty> serializer;
 
     InputPropertiesSerializer(ClassLoader classloader) {
-        this.serializer = new MapSerializer<String, Object>(BaseSerializerFactory.STRING_SERIALIZER, new DefaultSerializer<Object>(classloader));
+        this.serializer = new MapSerializer<String, InputProperty>(BaseSerializerFactory.STRING_SERIALIZER, new DefaultSerializer<InputProperty>(classloader));
```

Comment:
we should not use default java serialization any more, as it's very, very inefficient. We simply need to serialize the hash and serialized bytes for each entry